app/docs/md/conventions/preflight.md

```diff
@@ -0,0 +1,103 @@
+---
+title: Preflight
+---
+
+
+The purpose of the preflight file is to supply default application state.
+It can also be used to customize data based on the request per route.
+
+Enhance looks for the preflight file in the root of your app.
+
+<doc-code filename="app/preflight.mjs">
+
+```bash
+app
+├── api ............... data routes
+│   └── index.mjs ..... override default preflight application state with api data
+└── preflight.mjs ..... pre-populate application state
+
+```
+</doc-code>
+
+The preflight function is passed the request object enabling you to customize data per requested route.
+API responses are merged with the default state returned from preflight allowing you to override default state with specific API data per request.
+
+### Basic example 
+
+<doc-code filename="app/preflight.mjs">
+
+  ```javascript
+  export default function Preflight ({ req }) {
+    return { /* ...Your data here */ }
+  }
+````
+</doc-code>
+
+
+
+### Example of setting the page title using preflight
+
+<doc-code filename="app/preflight.mjs">
+
+  ```javascript
+  export default function Preflight ({ req }) {
+    return {
+      pageTitle: getPageTitle(req.path),
+      account: {
+        username: 'bobsyouruncle',
+        id: '23jk24h24'
+      }
+    }
+  }
+
+  function getPageTitle (path) {
+    const titleMap = {
+      '/': 'Home',
+      '/about': 'About',
+      '/account': 'My Account'
+    }
+
+    return titleMap[path] || 'My App Name'
+  }
+````
+</doc-code>
+
+
+
+### Example of overriding default preflight data with an API response
+
+<doc-code filename="app/preflight.mjs">
+
+  ```javascript
+  export default function Preflight ({ req }) {
+    return {
+      pageTitle: getPageTitle(req.path),
+      account: {
+        username: 'bobsyouruncle',
+        id: '23jk24h24'
+      }
+    }
+  }
+````
+</doc-code>
+
+<doc-code filename="app/api/index.mjs">
+
+  ```javascript
+  export async function get() {
+    return {
+      json: {
+        account: {
+          username: 'thisshouldoverride',
+          id: '39nr34n2'
+        }
+      }
+    }
+  }
```

Comment:
Wondering if the following example might demonstrate overrides a little more practically — it seems a bit odd to override an account on a certain route, but using the account to update the page title on the account route might be a bit more expected?

```suggestion
  export async function get() {
    return {
      json: {
        pageTitle: 'My Account (bobsyouruncle)',
        account: {
          username: 'bobsyouruncle',
          id: '23jk23h23'
        }
      }
    }
  }
```

Reply:
No I think this is how it should be shown.
The page title is handled in the preflight differently and I don't want to confuse people.
I needed to show a default that was overidden.
Imagine using preflight to mock out an account so development can progress then overriding it when the actual API is impelmented.

Reply:
Yeah, setting page title differently could be confusing, agreed. I was a bit thrown off by changing the account based on the path but I don't have any other ideas coming to mind! 😄